worker/sshserver/worker.go

```diff
@@ -94,30 +126,34 @@ func (ssw *serverWrapperWorker) loop() error {
 		return errors.Trace(err)
 	}
 
-	controllerConfig := ssw.config.SystemState.WatchControllerConfig()
-	if err := ssw.catacomb.Add(controllerConfig); err != nil {
+	controllerConfigWatcher := ssw.config.SystemState.WatchControllerConfig()
+	if err := ssw.catacomb.Add(controllerConfigWatcher); err != nil {
 		return errors.Trace(err)
 	}
 
-	changesChan := controllerConfig.Changes()
+	changesChan := controllerConfigWatcher.Changes()
 	for {
 		select {
 		case <-ssw.catacomb.Dying():
 			return ssw.catacomb.ErrDying()
 		case <-changesChan:
-			// TODO(ale8k): Once the configuration PR is merged, get the max conns & port
-			// from controller config. Get the HostKey from server info, and feed them through
-			// to NewServerWorker.
-
 			// Restart the server worker.
 			srv.Kill()
 			if err := srv.Wait(); err != nil {
 				return errors.Trace(err)
 			}
 
+			// Get latest controller configuration.
+			port, _, err := ssw.getLatestControllerConfig()
+			if err != nil {
+				return errors.Trace(err)
+			}
+
 			// Start the server again.
 			srv, err = ssw.config.NewServerWorker(ServerWorkerConfig{
-				Logger: ssw.config.Logger,
+				Logger:      ssw.config.Logger,
+				JumpHostKey: jumpHostKey,
+				Port:        port,
 			})
 			if err != nil {
 				return errors.Trace(err)
```

Comment:
Below this we re-add the server to the catacomb but we I don't see where we remove the server from the catacomb after killing it. Does that happen automatically? Or perhaps it's not necessary to remove it?

Reply:
Workers including tombs and catacombs work on a "kill kills all" ideology, as such when a tomb routine or worker within a catacomb die either from Kill() or independently it is removed (because it returns in the run func).